src/app/inventory/store/d2-item-factory.ts

```diff
@@ -528,6 +529,18 @@ export function makeItem(
     }
   }
 
+  // a weapon can have an inherent breaker type, or gain one from socketed mods
+  // (or armor mods can sort of add them but let's not go there quite yet)
+  // this is presented as an else-type dichotomy here, but who knows what the future holds
+  if (itemDef.breakerTypeHash)
+    createdItem.breakerType = defs.BreakerType.get(itemDef.breakerTypeHash);
+  else if (createdItem.bucket.inWeapons && createdItem.sockets) {
+    const breakerTypeHash = createdItem.sockets.sockets.find(
+      (s) => s.plug?.plugItem.breakerTypeHash
+    )?.plug?.plugItem.breakerTypeHash;
```

Comment:
Nice, this is a fallback for when the item data isn't working? I think `ItemComponent` should also have a live view of the breaker type for the item.

Reply:
there's potentially some bugs reported here https://github.com/Bungie-net/api/issues/1264
and breaker exotics don't have breaker in their instance info.

so since this needs to be split between inherent (for exotics) and mod-applied, it felt easiest to simply use the mod's presence.